modules/Emails/mail.php

```diff
@@ -467,6 +467,23 @@ function addAttachment($mail, $filename, $record) {
 	}
 }
 
+/** Function to check if a number is a attachmentid
+  * $id -- number or attachmentid
+  */
+function is_attachmentid($id) {
+	global $adb;
+	$adb->println('> is_attachment '.$id);
+	$result = $adb->pquery('SELECT attachmentsid FROM vtiger_attachments
+		INNER JOIN vtiger_crmentity ON attachmentsid=crmid WHERE deleted=0 AND attachmentsid=?', array(
+		$id
+	));
+	if ($adb->num_rows($result) > 0) {
```

Comment:
use a direct return here, it is a little faster:

`return ($adb->num_rows($result) > 0);`

you will have to move the "finished" debug message before the return